addons/microsoft_calendar/tests/test_sync_microsoft2odoo.py

```diff
@@ -221,3 +222,17 @@ def test_microsoft_recurrence_delete(self):
         events = self.env['calendar.event'].browse(event_ids).exists()
         self.assertFalse(recurrence, "It should remove recurrence")
         self.assertFalse(events, "It should remove all events")
+
+    def test_attendees_must_have_email(self):
+        MicrosoftCal = MicrosoftCalendarService(self.env['microsoft.service'])
+        partner = self.env['res.partner'].create({
+            'name': 'SuperPartner',
+        })
+        event = self.env['calendar.event'].create({
+            'name': "SuperEvent",
+            'start': datetime(2020, 3, 16, 11, 0),
+            'stop': datetime(2020, 3, 16, 13, 0),
+            'partner_ids': [(4, partner.id)],
+        })
+        with self.assertRaises(ValidationError):
+            event._sync_odoo2microsoft(MicrosoftCal)
```

Comment:
I would simply add a description for the test saying:
"""Synching with a partner without mail raises a ValidationError because Microsoft don't accept attendees without one."""